airflow/operators/python.py

```diff
@@ -255,6 +255,16 @@ def task(
     :type multiple_outputs: bool
 
     """
+    # try to infer from  type annotation
+    if python_callable and multiple_outputs is None:
+        sig = signature(python_callable).return_annotation
+        ttype = getattr(sig, "__origin__", None)
+
+        if sig != inspect.Signature.empty and ttype in (dict, Dict):
+            multiple_outputs = True
+
+    # in case its still none, then set False here
+    multiple_outputs = multiple_outputs or False
```

Comment:
Should we be able to skip this line? `None` has boolean value of `False`

Reply:
I'll do the patch suggestion you have above, removing the if statement would set it to false anyways so yeah the line is not needed